src/System.Runtime/tests/System/Reflection/AssemblyTests.cs

```diff
@@ -261,55 +260,116 @@ public static void GetManifestResourceStream()
         public static void Test_GlobalAssemblyCache()
         {
             Assert.False(typeof(AssemblyTests).Assembly.GlobalAssemblyCache);
-        }        
+        }
 
         [Fact]
         public static void Test_HostContext()
         {
             Assert.Equal(0, typeof(AssemblyTests).Assembly.HostContext);
-        }        
+        }
 
         [Fact]
         public static void Test_IsFullyTrusted()
         {
             Assert.True(typeof(AssemblyTests).Assembly.IsFullyTrusted);
-        }        
+        }
 
         [Fact]
-        public static void Test_SecurityRuleSet()
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "The full .NET Framework supports SecurityRuleSet")]
+        public static void Test_SecurityRuleSet_Netcore()
         {
             Assert.Equal(SecurityRuleSet.None, typeof(AssemblyTests).Assembly.SecurityRuleSet);
-        }        
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "SecurityRuleSet is ignored in .NET Core")]
+        public static void Test_SecurityRuleSet_Netfx()
+        {
+            Assert.Equal(SecurityRuleSet.Level2, typeof(AssemblyTests).Assembly.SecurityRuleSet);
+        }
 
         [Fact]
         public static void Test_LoadFile()
         {
-            var assem = typeof(AssemblyTests).Assembly;
-            string path = "System.Runtime.Tests.dll";
-            string fullpath = Path.GetFullPath(path);
+            Assembly currentAssembly = typeof(AssemblyTests).Assembly;
+            const string RuntimeTestsDll = "System.Runtime.Tests.dll";
+            string fullRuntimeTestsPath = Path.GetFullPath(RuntimeTestsDll);
+
+            var loadedAssembly1 = Assembly.LoadFile(fullRuntimeTestsPath);
+            if (PlatformDetection.IsFullFramework)
+            {
+                Assert.Equal(currentAssembly, loadedAssembly1);
+            }
+            else
+            {
+                Assert.NotEqual(currentAssembly, loadedAssembly1);
+            }
+
+            string dir = Path.GetDirectoryName(fullRuntimeTestsPath);
+            fullRuntimeTestsPath = Path.Combine(dir, ".", RuntimeTestsDll);
+
+            Assembly loadedAssembly2 = Assembly.LoadFile(fullRuntimeTestsPath);
+            if (PlatformDetection.IsFullFramework)
+            {
+                Assert.NotEqual(loadedAssembly1, loadedAssembly2);
```

Comment:
This is interesting to know that adding "." in the path causes Assembly.LoadFile to load again on desktop.